### PR TITLE
[RCM] fix(tracer_predicates): Match new tracer predicates constraints

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
To match the backend predicates logic, we want the tracer predicates to
follow the following rules:
- An empty list matches nothing
- At least one predicate matching is enough to pass

### Motivation
Preparing new product launches

### Additional Notes
There is a route in our internal API that will let you test this

### Possible Drawbacks / Trade-offs
Configurations currently using tracer predicates may see their behavior change. Still, as far as we know, this field is not used yet so the change shouldn't have much impact

### Describe how to test/QA your changes
- Create a configuration with a tracer predicate (or better: multiple tracer predicates) and make sure it lands where it should
- Create a configuration with an empty list of tracer predicates and make sure no one can access it
- Create a configuration with a list of an empty tracer predicate (`[{}]`) and make sure everyone can access it

### Reviewer's Checklist
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x]  At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [x] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [x] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [x] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [x] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/remote/service/client_predicates.go [hunk 1]
```diff
@@ -112,15 +111,14 @@ func executePredicate(client *pbgo.Client, predicates []*pbgo.TracerPredicateV1)
 				}
 
 				matched, errs := versionConstraint.Validate(version)
-				if len(errs) != 0 {
-					return false, fmt.Errorf("errors: %s", errs)
-				}
+				// We don't return on error here, it's simply the version not matching
 				if !matched || len(errs) > 0 {
-					return false, nil
+					continue
```
Comment (coignetp) [inline pkg/config/remote/service/client_predicates.go:116]: **🥜 nitpick:** ‏Can you add a test for this path?

File: pkg/config/remote/service/client_predicates.go [hunk 2]
```diff
@@ -112,15 +111,14 @@ func executePredicate(client *pbgo.Client, predicates []*pbgo.TracerPredicateV1)
...
 				}
 			}
 		}
+		return true, nil
```
Comment (coignetp) [inline pkg/config/remote/service/client_predicates.go:120]: **🥜 nitpick:** ‏Can you add a test for this path?

**Review by coignetp — APPROVED** [review]

File: pkg/config/remote/service/client_predicates.go (hunk 1 above)
Comment (BaptisteFoy) [reply pkg/config/remote/service/client_predicates.go:116]: There is one [here](https://github.com/DataDog/datadog-agent/pull/12980/files#diff-4ff734e21cc66f971c7e2118dee491340e10c87f708d828dfd1306a4a602a0eeR82) already

File: pkg/config/remote/service/client_predicates.go (hunk 2 above)
Comment (BaptisteFoy) [reply pkg/config/remote/service/client_predicates.go:120]: Done in [1fe4a7c](https://github.com/DataDog/datadog-agent/pull/12980/commits/1fe4a7cc27a30b1709ffa6f2efc37fdfb59022b3)

**Pull request merged**